src/Altinn.Apps/AppFrontend/react/altinn-app-frontend/src/components/base/DropdownComponent.test.tsx

```diff
@@ -137,4 +138,38 @@ describe('components/base/DropdownComponent', () => {
 
     expect(screen.queryByTestId('altinn-spinner')).not.toBeInTheDocument();
   });
+
+  it('should present replaced label if setup with values from repeating group in redux and trigger handleDataChanged with replaced values', () => {
+    const handleDataChange = jest.fn();
+
+    const dropdownWithOptionsFromRedux: IDropdownProps = {
+      id: 'component-id',
+      formData: {},
+      handleDataChange,
+      getTextResourceAsString: (value) => value,
+      readOnly: false,
+      isValid: true,
+      source: {
+        group: "someGroup",
+        label: "option.from.rep.group.label",
+        value: "someGroup[{0}].valueField"
+      },
+      ...({} as IComponentProps),
+    };
+    renderWithProviders(<DropdownComponent {...dropdownWithOptionsFromRedux} />, {
```

Comment:
Would it not be possible to use the `render` helper method here? Could extend it to accept a custom preloaded state as well. That would reduce some boilerplate, f.ex you would only need to pass overridden props to the component.